### ISSUE TITLE
v0.6.3 does not work

### ISSUE THREAD

**Issue by majst01** [metal-stack/os-metal-extension#29]
```

dener-extension-os-metal-758469f698-tszf6 gardener-extension-os-metal E0830 13:20:53.020216       1 reflector.go:140] k8s.io/client-go@v11.0.1-0.20190409021438-1a26190bd76a+incompatible/tools/cache/reflector.go:169: Failed to watch *v1.Lease: failed to list *v1.Lease: l
eases.coordination.k8s.io is forbidden: User "system:serviceaccount:extension-os-metal-nv7gt:gardener-extension-os-metal" cannot list resource "leases" in API group "coordination.k8s.io" at the cluster scope                                                                  
gardener-extension-os-metal-758469f698-tszf6 gardener-extension-os-metal {"level":"error","ts":"2023-08-30T13:21:04.145Z","msg":"Could not wait for Cache to sync","controller":"operatingsystemconfig","error":"failed to wait for operatingsystemconfig caches to sync: timed o
ut waiting for cache to be synced","stacktrace":"sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2.1\n\tsigs.k8s.io/controller-runtime@v0.13.0/pkg/internal/controller/controller.go:210\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(
*Controller).Start.func2\n\tsigs.k8s.io/controller-runtime@v0.13.0/pkg/internal/controller/controller.go:215\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start\n\tsigs.k8s.io/controller-runtime@v0.13.0/pkg/internal/controller/controller.go:241\nsig
s.k8s.io/controller-runtime/pkg/manager.(*runnableGroup).reconcile.func1\n\tsigs.k8s.io/controller-runtime@v0.13.0/pkg/manager/runnable_group.go:219"}                                                                                                                           
gardener-extension-os-metal-758469f698-tszf6 gardener-extension-os-metal {"level":"info","ts":"2023-08-30T13:21:04.145Z","msg":"Stopping and waiting for non leader election runnables"}                                                                                         
gardener-extension-os-metal-758469f698-tszf6 gardener-extension-os-metal {"level":"info","ts":"2023-08-30T13:21:04.145Z","msg":"Stopping and waiting for leader election runnables"}                                                                                             
gardener-extension-os-metal-758469f698-tszf6 gardener-extension-os-metal {"level":"info","ts":"2023-08-30T13:21:04.145Z","msg":"Shutdown signal received, waiting for all workers to finish","controller":"heartbeat"}                                                           
gardener-extension-os-metal-758469f698-tszf6 gardener-extension-os-metal {"level":"error","ts":"2023-08-30T13:21:04.145Z","msg":"Reconciler error","controller":"heartbeat","object":{"name":""},"namespace":"","name":"","reconcileID":"e856071d-81bb-4c46-bd15-38660f767f7b","e
rror":"Timeout: failed waiting for *v1.Lease Informer to sync","stacktrace":"sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler\n\tsigs.k8s.io/controller-runtime@v0.13.0/pkg/internal/controller/controller.go:326\nsigs.k8s.io/controller-ru
ntime/pkg/internal/controller.(*Controller).processNextWorkItem\n\tsigs.k8s.io/controller-runtime@v0.13.0/pkg/internal/controller/controller.go:273\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2.2\n\tsigs.k8s.io/controller-runtime@v0.13.0
/pkg/internal/controller/controller.go:234"}                                                                                                                                                                    

```
I guess we missed some RBAC for Lease listing

**Comment by majst01** [comment]
RBAC is fixed with #30 but now:

```
gardener-extension-os-metal-55dcb88f84-krrrm gardener-extension-os-metal {"level":"error","ts":"2023-08-30T14:03:00.966Z","msg":"Reconciler error","controller":"heartbeat","object":{"name":""},"namespace":"","name":"","reconcileID":"1afe7040-052b-4c50-8aea-4d26d011c162","error":"an empty namespace may not be set during creation","stacktrace":"sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler\n\tsigs.k8s.io/controller-runtime@v0.13.0/pkg/internal/controller/controller.go:326\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem\n\tsigs.k8s.io/controller-runtime@v0.13.0/pkg/internal/controller/controller.go:273\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2.2\n\tsigs.k8s.io/controller-runtime@v0.13.0/pkg/internal/controller/controller.go:234"}
```

**Comment by majst01** [comment]
With latest commit in #30 the heartbeat controller also works
